### PR TITLE
Student feedback results sorts by recipient incorrectly #3780

### LINKED ISSUE #3780: Student feedback results sorts by recipient incorrectly
Responses for the same recipient from different givers are shown in different tables, whereas they should   be in the same table

### PR DESCRIPTION
**Pull request by taniach**
Fixes #3780

### CODE REVIEW

File: src/main/java/teammates/common/datatransfer/FeedbackSessionResultsBundle.java [hunk 1]
```diff
@@ -1049,6 +1049,31 @@ public String getQuestionText(String feedbackQuestionId) {
...
+        Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> sortedMap =
+                new TreeMap<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>();
+
+        for (FeedbackQuestionAttributes question : questions.values()) {
+            sortedMap.put(question, new ArrayList<FeedbackResponseAttributes>());
+        }
+
+        for (FeedbackResponseAttributes response : responses) {
+            List<FeedbackResponseAttributes> responsesForQuestion = sortedMap
+                    .get(questions.get(response.feedbackQuestionId));
```
Comment (JosephineKwa) [inline src/main/java/teammates/common/datatransfer/FeedbackSessionResultsBundle.java]: Perhaps you could extract `response.feedbackQuestionId` into a local variable to make more readable

**Comment by JosephineKwa** [comment]
I seem to be failing StudentFeedbackResultsPageUiTest at line 109. I think the team response from Charlie should come before Danny

**Comment by JosephineKwa** [comment]
Some changes requested

**Comment by taniach** [comment]
That test is passing on my computer. Dev green

**Comment by taniach** [comment]
@JosephineKwa ready for review

**Pull request merged**